### ISSUE TITLE
Not updating temperature or boost

### ISSUE THREAD

**Issue by skoviss** [Pyhass/Hive-Custom-Component#71]
The Hive Thermostat Boost state is not updating if I turn it on my phone or on my Hive Thermostat.
Updating unless I reload the home assistant.

**Comment by AndrewGatenbyVS** [comment]
I get similar problems, where the status and temperature from my Hive thermostat is only updated when the integration is reloaded via Configuration > Integrations > Hive > ... > Reload. The Current Temperature doesn't vary in HA, even though the value does vary and shows up in the actual Hive app.

The logs show similar messages to this for various Hive entities:

```
Logger: homeassistant.components.sensor
Source: custom_components/hive/sensor.py:189
Integration: Sensor (documentation, issues)
First occurred: 4 April 2021, 01:23:18 (3 occurrences)
Last logged: 00:24:39
hive: Error on device update!

Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 360, in _async_add_entity
    await entity.async_device_update(warning=False)
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 465, in async_device_update
    await task
  File "/config/custom_components/hive/sensor.py", line 135, in async_update
    self.attributes = await self.get_current_temp_sa()
  File "/config/custom_components/hive/sensor.py", line 189, in get_current_temp_sa
    if temperature_target > temp_current:
TypeError: '>' not supported between instances of 'NoneType' and 'NoneType'
```

```
Logger: homeassistant.components.climate
Source: custom_components/hive/climate.py:99
Integration: Climate (documentation, issues)
First occurred: 4 April 2021, 01:23:19 (6 occurrences)
Last logged: 00:24:39

    Error adding entities for domain climate with platform hive
    Error while setting up hive platform for climate

Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 317, in async_add_entities
    await asyncio.gather(*tasks)
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 385, in _async_add_entity
    device_info = entity.device_info
  File "/config/custom_components/hive/climate.py", line 99, in device_info
    "model": self.device["deviceData"]["model"],
KeyError: 'model'
```

If it helps, I'm running HA 2021.3.4 via Docker on a Raspberry Pi 3 host.

**Comment by KJonline** [comment]
For model error. Please can you link to the version of the hive thermostat that you have and your setup e.g wired in or not

**Comment by AndrewGatenbyVS** [comment]
@KJonline I've got the Hive 2 Wireless Thermostat for combi boilers (first one on this page https://community.hivehome.com/s/article/Which-thermostat-do-I-have). I cant tell a specific model number - there's nothing obvious on the thermostat or receiver, or available via the app sorry.

**Comment by KJonline** [comment]
Is this still an issue?

**Comment by skoviss** [comment]
Yes
App
![hive 2](https://user-images.githubusercontent.com/44526247/128603586-ebfcbc50-60f5-4ac7-8472-f70fe3aec649.jpg)
Home assistant
![image_2021-08-07_153056](https://user-images.githubusercontent.com/44526247/128603587-c6d492d7-c839-4bc7-807e-2f57016cee7f.png)
Not updating when your changing temperature or turning on boost mode in a app!

**Comment by upstairs-at-erics** [comment]
same here on the combi version of the thermostat/receiver

**Comment by g60** [comment]
Hi, was anyone able to resolve this, I'm having a similar issue with my system except the error states KeyError: 'manufacturer'

**Comment by upstairs-at-erics** [comment]
no, still only only updates once on startup for my receiver type. There are two models of receiver one for F/E system and one for combi boilers.  My hunch is that the combi boiler one is subtly different and not being handled in the code, hence the keys in error for manufacturer and model. But thats just an uneducated guess,

**Comment by g60** [comment]
I think I've managed to get mine working again now. The issue was caused by having a thermostat from our old house still showing in the hive account (we moved and took our existing hub with us and bought 2 new thermostats for the new house but forgot to remove the old thermostat before we moved), I had to contact Hive to get them to remove the old one from my account. Once they did that I removed the integration and re-added it as that was the only way to remove it from HA.

I'm not sure if any of that helps you with your issue, I think mine was probably quite unique but thought I'd let you know in case it gives you any idea's of things to check. Also our old house had a combi boiler and the new house has a gravity fed boiler so I don't think it's an incompatibility with combi boilers.
